client/pool.go

```diff
@@ -0,0 +1,46 @@
+package client
+
+import (
+	"context"
+	"time"
+
+	"github.com/micro/go-micro/transport"
+)
+
+// Pool is a connection pool.
+type Pool interface {
+	Conn(ctx context.Context, addr string) Conn
+	Release(ctx context.Context, addr string, conn Conn, err error)
```

Comment:
Reorder alphabetically to honour the alphabet Gods!

Reply:
[feature/client-pool-interface ace08a9] ABCDEFG....